cherrypy/lib/encoding.py

```diff
@@ -412,7 +415,13 @@ def gzip(compress_level=5, mime_types=['text/html', 'text/plain'],
                 cherrypy.log('Gzipping', context='TOOLS.GZIP')
             # Return a generator that compresses the page
             response.headers['Content-Encoding'] = 'gzip'
-            response.body = compress(response.body, compress_level)
+            # Check if there exists a pre-compressed file, and use it.
+            file = cherrypy.serving.request.path_abs + '.gz'
+            if os.access(file, os.F_OK):
+                from cherrypy.lib.static import serve_file
```

Comment:
I'll have to check this later. I'm not completely following you, OTOH, I'm not at a computer for the moment.